.github/workflows/ci.yml

```diff
@@ -0,0 +1,167 @@
+name: New CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  # Pre-checks
+  format-rust:
+    name: Rust code Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rustfmt
+          override: true
+      - name: rustfmt
+        run: cargo +nightly fmt --check
+
+  clippy-rust:
+    name: Clippy lint
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup component add clippy
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets
+
+  # Builds
+  build-node-binaries:
```

Comment:
another shortcut we could do is build conditionally iif there are changes in the node code /libs

Reply:
It's hard to fit in this model, since it would mean downloading the node from another cache. And compiling the node is actually faster now than installing all python dependencies, so it basically costs no additional time (they run in parallel).

Also, the alternative to avoid compiling the node is downloading the binaries from the main branch. This is tricky and we already had a problem in the past with it, since current main can be different from the point this pr branched off of main, which is what we would actually want.

I think this is better for simplicity. As a bonus, it avoids needing to maintain that list of which directories belong to the node vs python :sweat_smile: .

Reply:
Sounds good. In fact, we ended up in the situation we are right now trying to optimize these kind of things too early.